### DISCUSSION TITLE
Is there a way to inject Commands into Hyde from a custom extension?

### DISCUSSION THREAD

**Discussion by raphaelstolt** [hydephp/hyde#266] [Q&A] [answered]
Can't find anything about this topic from the documentation.

**Comment by emmadesilva** [comment]
Hi! The best way is probably to do it from a service provider just like in Laravel! I can send you some links with examples next time I'm on the computer?

**Reply by emmadesilva** [reply, marked as answer]
@raphaelstolt We register the Hyde commands in [`ConsoleServiceProvider.php`](https://github.com/hydephp/framework/blob/d5b2f5c547d44f83e39bc6e58c130af3b3f618e8/src/Console/ConsoleServiceProvider.php), so that's a great option if you are developing an extension.

The rest of the [Laravel Artisan Console](https://laravel.com/docs/10.x/artisan) applies, and you can also use the underlying Symfony helpers if you need more control.

If this answers your question, feel free to mark it as an answer! Let me know if you need any more help! By the way, I'd love to see your extension when it's done!
